test/models/report_test.rb

```diff
@@ -14,7 +14,8 @@ class ReportTest < ActiveSupport::TestCase
 
   test 'created_on' do
     alice_report = reports(:alice)
-    assert_equal Date.current, alice_report.created_on
+    alice_report.created_at = Time.zone.local(2024, 10, 9, 0, 0, 0)
+    assert_equal Date.new(2024, 10, 9), alice_report.created_on
```

Comment:
```suggestion
    alice_report.created_at = "2024-10-09 10:30".in_time_zone
    assert_equal "2024-10-09".to_date, alice_report.created_on
```

好みによるところも多いかもしれませんが、個人的には文字列をDateやTimeに変換する方が読みやすくて好きですね。
あと、時刻部分は無視されることを確認するために、あえて00:00以外の時間を指定するといいかもしれません